### ISSUE TITLE
No warnings produced for incorrect selector format

### ISSUE THREAD

**Issue by gunnx** [davidtheclark/stylelint-selector-bem-pattern#22]
Have setup plugin as per docs
Decided to override BEM format to change modifiers to `--` instead of `_`

When I put an incorrect selector and run stylelint over it, no warnings are produced. 

`stylelint` version: 7.6.0
`stylelint-selector-bem-pattern` version: 1.0.0

Test SCSS file:
```
.component_____block----modifier
```

Rules: 
```
'use strict';

function bemSelectors(block, presetOptions) {
  //const ns = (presetOptions && presetOptions.namespace) ? presetOptions.namespace + '-' : '';
  const WORD = '[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*';
  const element = '(?:__' + WORD + ')?';
  const modifier = '(?:--' + WORD + '){0,2}';
  const attribute = '(?:\\[.+\\])?';
  return new RegExp('^\\.' + block + element + modifier + attribute + '$');
}

module.exports = {
  plugins: [
   'stylelint-scss',
   'stylelint-selector-bem-pattern'
  ],
  rules: {
    'at-rule-empty-line-before': [ 'always', {
      except: [
        'blockless-after-same-name-blockless',
        'first-nested',
      ],
      ignore: ['after-comment'],
    } ],
    'at-rule-name-case': 'lower',
    'at-rule-name-space-after': 'always-single-line',
    'at-rule-semicolon-newline-after': 'always',
    'block-closing-brace-empty-line-before': 'never',
    'block-closing-brace-newline-after': 'always',
    'block-closing-brace-newline-before': 'always-multi-line',
    'block-closing-brace-space-before': 'always-single-line',
    'block-no-empty': true,
    'block-opening-brace-newline-after': 'always',
    'block-opening-brace-space-after': 'never-single-line',
    'block-opening-brace-space-before': 'always',
    'color-hex-case': 'lower',
    'color-hex-length': 'short',
    'color-no-invalid-hex': true,
    'comment-empty-line-before': [ 'always', {
      except: ['first-nested'],
      ignore: ['stylelint-commands'],
    } ],
    'comment-no-empty': true,
    'comment-whitespace-inside': 'always',
    'custom-property-empty-line-before': [ 'always', {
      except: [
        'after-custom-property',
        'first-nested',
      ],
      ignore: [
        'after-comment',
        'inside-single-line-block',
      ],
    } ],
    'declaration-bang-space-after': 'never',
    'declaration-bang-space-before': 'always',
    'declaration-block-no-duplicate-properties': [ true, {
      ignore: ['consecutive-duplicates-with-different-values'],
    } ],
    'declaration-block-no-redundant-longhand-properties': true,
    'declaration-block-no-shorthand-property-overrides': true,
    'declaration-block-semicolon-newline-after': 'always',
    'declaration-block-semicolon-space-after': 'always-single-line',
    'declaration-block-semicolon-space-before': 'never',
    'declaration-block-single-line-max-declarations': 1,
    'declaration-block-trailing-semicolon': 'always',
    'declaration-colon-newline-after': 'always-multi-line',
    'declaration-colon-space-after': 'always-single-line',
    'declaration-colon-space-before': 'never',
    'declaration-empty-line-before': [ 'always', {
      except: [
        'after-declaration',
        'first-nested',
      ],
      ignore: [
        'after-comment',
        'inside-single-line-block',
      ],
    } ],
    'font-family-no-duplicate-names': true,
    'function-calc-no-unspaced-operator': true,
    'function-comma-newline-after': 'always-multi-line',
    'function-comma-space-after': 'always-single-line',
    'function-comma-space-before': 'never',
    'function-linear-gradient-no-nonstandard-direction': true,
    'function-max-empty-lines': 0,
    'function-name-case': 'lower',
    'function-parentheses-newline-inside': 'always-multi-line',
    'function-parentheses-space-inside': 'never-single-line',
    'function-whitespace-after': 'always',
    'indentation': 2,
    'keyframe-declaration-no-important': true,
    'length-zero-no-unit': true,
    'max-empty-lines': 1,
    'media-feature-colon-space-after': 'always',
    'media-feature-colon-space-before': 'never',
    'media-feature-name-case': 'lower',
    'media-feature-name-no-unknown': true,
    'media-feature-parentheses-space-inside': 'never',
    'media-feature-range-operator-space-after': 'always',
    'media-feature-range-operator-space-before': 'always',
    'media-query-list-comma-newline-after': 'always-multi-line',
    'media-query-list-comma-space-after': 'always-single-line',
    'media-query-list-comma-space-before': 'never',
    'no-empty-source': true,
    'no-eol-whitespace': true,
    'no-extra-semicolons': true,
    'no-invalid-double-slash-comments': true,
    'no-missing-end-of-source-newline': true,
    'number-leading-zero': 'never',
    'number-no-trailing-zeros': true,
    'property-case': 'lower',
    'property-no-unknown': true,
    'rule-nested-empty-line-before': [ 'always-multi-line', {
      except: ['first-nested'],
      ignore: ['after-comment'],
    } ],
    'rule-non-nested-empty-line-before': [ 'always-multi-line', {
      ignore: ['after-comment'],
    } ],
    'selector-attribute-brackets-space-inside': 'never',
    'selector-attribute-operator-space-after': 'never',
    'selector-attribute-operator-space-before': 'never',
    'selector-combinator-space-after': 'always',
    'selector-combinator-space-before': 'always',
    'selector-descendant-combinator-no-non-space': true,
    'selector-list-comma-newline-after': 'always',
    'selector-list-comma-space-before': 'never',
    'selector-max-empty-lines': 0,
    'selector-pseudo-class-case': 'lower',
    'selector-pseudo-class-no-unknown': true,
    'selector-pseudo-class-parentheses-space-inside': 'never',
    'selector-pseudo-element-case': 'lower',
    'selector-pseudo-element-colon-notation': 'double',
    'selector-pseudo-element-no-unknown': true,
    'selector-type-case': 'lower',
    'selector-type-no-unknown': true,
    'shorthand-property-no-redundant-values': true,
    'string-no-newline': true,
    'unit-case': 'lower',
    'unit-no-unknown': true,
    'value-list-comma-newline-after': 'always-multi-line',
    'value-list-comma-space-after': 'always-single-line',
    'value-list-comma-space-before': 'never',
    'value-list-max-empty-lines': 0,
    'scss/at-extend-no-missing-placeholder': true,
    'scss/selector-no-redundant-nesting-selector': true,
    'scss/at-mixin-argumentless-call-parentheses': 'never',
    'plugin/selector-bem-pattern': {
      componentName: '/^[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*$/',
      componentSelectors: bemSelectors,
      ignoreSelectors: '/^\.no-.+$/'
    }
  }
};
```

**Comment by davidtheclark** [comment]
@gunnx Can you confirm that the regular expression you end up with *does* reject the selector? Trying to determine if it's a problem with your regular expression or with this package.

**Comment by Kikobeats** [comment]
I'm trying now and yes, it doesn't work

**Comment by davidtheclark** [comment]
> it doesn't work

@Kikobeats: What doesn't work, when?

**Comment by Kikobeats** [comment]
I run the example with intentionally bad code to see the linter in action but nothing happens :(

**Comment by davidtheclark** [comment]
@Kikobeats Sorry, but I have no idea what you're talking about. Are you talking about the regular expression described above? Did you test if it works *outside* of this package? Or are do you have your own issue?

**Comment by Kikobeats** [comment]
@davidtheclark 

This should be produce an error according with tests:

```
echo ".Foo_bar {}" | ./node_modules/.bin/stylelint --config .stylelintrc
```

but nothing happens

**Comment by davidtheclark** [comment]
@Kikobeats: If you can provide a simplified reduced test case, we could start from there. Please open another issue if you can.

**Comment by Kikobeats** [comment]
more reduce than use the CLI? sorry but I don't understand your point.

**Comment by davidtheclark** [comment]
@Kikobeats, if you're using the config above — there is a lot going on there. Reduce it to the essentials. Also, did you try postcss-bem-linter alone? Are you sure your regex works even outside of these tools? These are essential debugging steps.

**Comment by Kikobeats** [comment]
My config used is the same as README suggest, removing the comments:

```json
// .stylelintrc
{
  "plugins": [
    "stylelint-selector-bem-pattern"
  ],
  "rules": {
    "plugin/selector-bem-pattern": {
      "componentName": "[A-Z]+",
      "componentSelectors": {
        "initial": "^\\.{componentName}(?:-[a-z]+)?$",
        "combined": "^\\.combined-{componentName}-[a-z]+$"
      },
      "utilitySelectors": "^\\.util-[a-z]+$"
    },
  }
}
```

**Comment by gunnx** [comment]
@Kikobeats have you remember to include the comments in the SASS files
```
/** @define block-name */
.block-name {}
.block-name__mod-name--val {}
/* @end */
```

**Comment by defel** [comment]
@gunnx came here for exactly the same problem, works with this line now

**Comment by gunnx** [comment]
@defel you mean the @define?

**Comment by greaveselliott** [comment]
Similar issue here, I am adding the opening and closing block definitions.

I am using the following stylelint rule:
```
    "plugin/selector-bem-pattern": {
      "componentName": "[a-z]+",
      "componentSelectors": {
        "initial": "^\\.{componentName}(?:-[a-z]+)?$",
        "combined": "^\\.combined-{componentName}-[a-z]+$"
      },
      "utilitySelectors": "^\\.util-[a-z]+$"
    }
```


I have a selector that references a mixin. 

```
.a-heading-one {
   @include a-heading-one();
}
```

If I break this BEM selector by adding an incorrect name, no error is thrown.
```
.a-heading-one_BrokenSelector {
   @include a-heading-one();
}
```

But if I add a static property to this broken selector, an error is thrown
```
.a-heading-one {
   @include a-heading-one();
   color: red;
}
```

Any ideas?

**Comment by davidtheclark** [comment]
@greaveselliott Do you have `/** @define block-name */` in your code?

If so and it's still broken, I suggest writing a test that demonstrates the failure.
